src/Microsoft.DotNet.Wpf/src/System.Xaml/System/Xaml/Schema/XamlTypeInvoker.cs

```diff
@@ -202,27 +189,28 @@ public virtual MethodInfo GetAddMethod(XamlType contentType)
             MethodInfo addMethod;
             if (_addMethods == null)
             {
-                Dictionary<XamlType, MethodInfo> addMethods = new Dictionary<XamlType, MethodInfo>();
+                var addMethods = new Dictionary<XamlType, MethodInfo>();
                 addMethods.Add(_xamlType.ItemType, _xamlType.AddMethod);
                 foreach (XamlType type in _xamlType.AllowedContentTypes)
                 {
-                    addMethod = CollectionReflector.GetAddMethod(
-                        _xamlType.UnderlyingType, type.UnderlyingType);
+                    addMethod = CollectionReflector.GetAddMethod(_xamlType.UnderlyingType, type.UnderlyingType);
                     if (addMethod != null)
                     {
-                        addMethods.Add(type, addMethod);
+                        // Use TryAdd as AllowedContentTypes can contain
+                        // duplicate types.
+                        addMethods.TryAdd(type, addMethod);
```

Comment:
are the add methods the same in this case?

Reply:
I would expect the add method to be the same when called on the same type, so yes this add method would be the same method for 
 the existing type as for the duplicate type